### PR TITLE
[fix] Convert BufferSlice to BufferLoad when used as range/loop start and end

### PR DESCRIPTION
**Pull request by yzh119**
A quick fix of the parser issue mentioned in #10327 .
Ranges and loops require `start` and `stop` to be PrimExpr, however, `BufferSlice` is not always scalar so it's not a `PrimExpr`.
This PR performs the transformation.

A deeper question is should we make `BufferSlice` a `PrimExpr`? 

CC @MasterJH5574 @Hzfengsy

### CODE REVIEW

File: tests/python/unittest/test_tvmscript_bufferslice_loop_range.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (MasterJH5574) [inline tests/python/unittest/test_tvmscript_bufferslice_loop_range.py:1]: I suggest moving this test to the round-trip test file? That makes sense to me and I think there's no need for a new file.

File: tests/python/unittest/test_tvmscript_bufferslice_loop_range.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from tvm.script import tir as T
+
+
+@T.prim_func
+def segment_sum(A_ptr: T.handle, B_ptr: T.handle, indptr_ptr: T.handle, n: T.int32, m: T.int32) -> None:
+    A = T.match_buffer(A_ptr, [m], dtype="float32")
+    B = T.match_buffer(B_ptr, [n], dtype="float32")
+    indptr = T.match_buffer(indptr_ptr, [n + 1], dtype="int32")
+    # body
+    # with T.block("root")
```
Comment (MasterJH5574) [inline tests/python/unittest/test_tvmscript_bufferslice_loop_range.py:27]: Heyy perhaps we can remove them.

**Review by MasterJH5574 — APPROVED** [review]
LGTM! Thank you Zihao!

> A deeper question is should we make `BufferSlice` a `PrimExpr`?

cc @spectrometerHBH @Hzfengsy. Could you guys reason about it :eyes: ?

File: tests/python/unittest/test_tvmscript_bufferslice_loop_range.py (hunk 1 above)
Comment (yzh119) [reply tests/python/unittest/test_tvmscript_bufferslice_loop_range.py]: DONE

**Review by junrushao1994 — APPROVED** [review]

**Comment by junrushao1994** [comment]
Thanks @yzh119!!

**Pull request merged**